actionpack/lib/abstract_controller/callbacks.rb

```diff
@@ -48,7 +48,8 @@ def _normalize_callback_options(options)
 
       def _normalize_callback_option(options, from, to) # :nodoc:
         if from = options[from]
-          from = Array(from).map {|o| "action_name == '#{o}'"}.join(" || ")
+          _from = Array(from).map(&:to_s)
```

Comment:
Hm, looks like creating Sets is too slow. [Here's a benchmark](https://gist.github.com/nateberkopec/43cb5cfa95a417b90485). With an Array of 10 elements, creating a Set and then calling #include? is 5x slower than creating an Array and calling #include?